File: Pascal'sTriangleII.cs
public class Solution {
    private void helper2(ref IList<int> rez, int i, int j = 1, int t = 0, int tmp = 1) {
        if(i + 1 == j) {
            return;
        }
        if(i == j) {
            rez[j] = 1;
        }
        rez[j-1] = tmp + t;
        helper2(ref rez, i, j+1, tmp, rez[j]);
    }
    
    private IList<int> helper(ref IList<int> rez, int i) {
        if(rez.Count == i) {
            return rez;
        }
        // int tmp = 1;
        // int t = 0;
        // for(int j=1; j<=i; j++) {
        //     if(i==j) {
        //         rez[j] = 1;
        //     }
        //     rez[j - 1] = tmp + t;
        //     t = tmp;
        //     tmp = rez[j];
        // }
        helper2(ref rez, i);
        helper(ref rez, i + 1);
        return rez;
    }
    
    public IList<int> GetRow(int rowIndex) {
        IList<int> rez = new List<int>(new int[rowIndex + 1]);
        rez[0] = 1;
        helper(ref rez, 0);
        return rez;
    }
}

File: Pascal'sTriangleII.cpp
#pragma GCC optimize("O3,unroll-loops")
#pragma GCC target("avx2,bmi,bmi2,lzcnt,popcnt")

static const bool __boost = [](){
    cin.tie( nullptr );
    cout.tie( nullptr );
    return ios_base::sync_with_stdio( false );
}();

class Solution {
public:
    vector<int> getRow(int rowIndex) {
        std::vector<int> rez(rowIndex+1, 0);
        rez[0] = 1;
        for(int i=1; i<=rowIndex; i++) {
            int tmp{1};
            for(int j=1; j<=i; j++) {
                if(j == i) {
                    rez[j] = 1;
                }
                int t = rez[j-1];
                rez[j-1] = tmp;
                tmp = rez[j] + t;
            }
        }
        return rez;
    }
};
